src/Agent.Plugins/RepositoryPlugin.cs

```diff
@@ -81,6 +82,31 @@ public override async Task RunAsync(AgentTaskPluginExecutionContext executionCon
             var repoAlias = executionContext.GetInput(Pipelines.PipelineConstants.CheckoutTaskInputs.Repository, true);
             var repo = executionContext.Repositories.Single(x => string.Equals(x.Alias, repoAlias, StringComparison.OrdinalIgnoreCase));
 
+            // Merge the repository checkout options
+            if ((!executionContext.Variables.TryGetValue("MERGE_CHECKOUT_OPTIONS", out VariableValue mergeCheckoutOptions) || !String.Equals(mergeCheckoutOptions.Value, "false", StringComparison.OrdinalIgnoreCase)) &&
```

Comment:
I haven't tested this change yet. getting early feedback.

Reply:
you might also want to check the message downgrade code and make sure everything still lines up. (AgentJobMessageUtil.cs)